### ISSUE TITLE
Why need to set frame size to multiple of 128?

### ISSUE THREAD

**Issue by DBraun** [leomccormack/SPARTA#55]
For ambiBIN, is it necessary to require the frame size to be a multiple of 128?

https://github.com/leomccormack/Spatial_Audio_Framework/blob/e32c6ec8f1e87fa6360aa75725b3073d02fef1c4/examples/src/ambi_bin/ambi_bin.c#L408 and https://github.com/leomccormack/SPARTA/blob/b696ecbd9c9323c47b61147e349a3f014afcf57b/audio_plugins/_SPARTA_ambiBIN_/src/PluginProcessor.cpp#L289

![2022-03-02 11_01_41-](https://user-images.githubusercontent.com/2096055/156432253-ddf45c1f-c896-442a-983d-70798b88f2ce.png)

I'm trying to use the plugin in a real-time environment, and I think it has a fluctuating non-standard block size. If the audio rate is 44100, and the video rate of the environment is 60, then the requested number of samples is 735, and this is not a multiple of 128.

**Comment by leomccormack** [comment]
Hi, apologies for the delayed response.
As the plugin is written now, the framesize of a multiple of 128 is strict. 

Note that this isn't the case for all SPARTA plugins. E.g. the sparta_powermap will accept any block sizes by feeding/reading from FIFO buffers. However, this incurs some additional latency. 
For e.g. sparta_ambiBIN, responsiveness was deemed to be more important than having the flexibility of the FIFO buffers. Here, the input 128 sample frames and processed right-away. Any excess samples are instead zeroed.

However, having both options would be ideal.... However, I've not managed to get around to this yet, so actually if anyone is willing to give this a go, then I can offer some guidance along the way : )

**Comment by mdedwards** [comment]
Hello, I think I'm getting this error when using the Panner VST. I can't be sure as it only flashes up for a fraction of a second every so often. I don't see anywhere to set frame size. Best, Michael

**Comment by lossius** [comment]
Am I right that this has now been resolved in version 1.8.2? In that case, thanks! If this allows using the plugins while listening over AirPods, it will be useful when working out of the studio. AirPods, for some reason, seem to use a non-2^n vector size. 

The release notes say that "the plugins will instead state what the optimal framesize is (in yellow)". What is the benefit of sticking to optimal frame sizes? Does it reduce processing demands, latency, or something else that is worth being aware of?

**Comment by jananifernandez** [comment]
Yes, this is now no longer a strict requirement. All plugins will work with any framesize and samplerate.
However, non multiples of framesizes will show this warning message 
`"Set host block size to \"") + String(ambi_bin_getFrameSize()) + TRANS("\" for lowest latency")"
`
Since other block sizes will incur extra buffering and delay.

Also, samplerates other than 44.1/48kHz, will show this warning message:
`"Sample rate \"") + String(ambi_bin_getDAWsamplerate(hAmbi)) + TRANS("\" is not recommended"`
This is because the DSP is optimised for 44.1/48kHz. In particular, the parametric plugins are tuned for these samplerates, and could behave differently for other samplerates.  Although, in many cases, they also still work well.
